.github/workflows/promote.yml

```diff
@@ -170,3 +170,202 @@ jobs:
           echo "### Branches" >> $GITHUB_STEP_SUMMARY
           echo "- **Source:** main" >> $GITHUB_STEP_SUMMARY
           echo "- **Target:** production" >> $GITHUB_STEP_SUMMARY
+
+  calculate-version:
+    runs-on: ubuntu-latest
+    needs: promote
```

Comment:
Technically we could release containers perfectly fine without first needing "promote" to succeed, but I think released containers should reflect what is on prod, so I'm adding this prerequisite in case "promote" fails. I'm open to the opposition.